client/src/main/java/tds/assessment/Item.java

```diff
@@ -137,10 +141,47 @@ public void setItemProperties(List<ItemProperty> itemProperties) {
     }
 
     /**
+     * @return The full path and file name of the item's XML file
+     */
+    public String getItemFilePath() {
+        return itemFilePath;
+    }
+
+    public void setItemFilePath(String itemFilePath) {
+        this.itemFilePath = itemFilePath;
+    }
+
+    /**
+     * @return The full path and file name of the item's stimulus file (if there is one)
+     */
+    public String getStimulusFilePath() {
+        return stimulusFilePath;
+    }
+
+    public void setStimulusFilePath(String stimulusFilePath) {
+        if (stimulusFilePath == null) {
```

Comment:
I would argue that we could make the `tblitem.filepath` non-nullable or at least treat it as such in the application, since I can't think of a possible way that a valid item does not have an item path since without one you can't display an item in the TDS.